d3a_api_client/rest_device.py

```diff
@@ -166,9 +166,20 @@ def _on_tick(self, message):
         self.callback_thread.submit(execute_function_util, function=function,
                                     function_name="on_tick")
 
-    def _on_trade(self, message):
-        logging.info(f"<-- {message.get('buyer')} BOUGHT {round(message.get('energy'),4)} kWh "
+    @staticmethod
+    def _log_trade_info(message):
+        logging.info(f"<-- {message.get('buyer')} BOUGHT {round(message.get('energy'), 4)} kWh "
                      f"at {round(message.get('price'), 2)}/kWh -->")
```

Comment:
So we are saying that `message.get('energy')` never returns None and skip the None check now?

Reply:
Just for clarification, I get the error fixing below.

Reply:
Forget it, yes, you added a new log message that should tell us directly what is happening for better debugging later.

Reply:
It indeed never returns None, and even if this happens at some point due to a d3a bug, the Exception will be caught and logged by the websocket_device.py -> received_message() and we'll get notified of both the traceback and the incoming message for debugging.